astroid/nodes/scoped_nodes/mixin.py

```diff
@@ -26,7 +27,7 @@ class LocalsDictNodeNG(node_classes.LookupMixIn):
 
     # attributes below are set by the builder module or by raw factories
 
-    locals: dict[str, list[nodes.NodeNG]] = {}
+    locals: dict[str, list[SuccessfulInferenceResult]] = {}
```

Comment:
Do we want to update all other places where we define `local`?

Reply:
Pushed an update. AFAICT there wasn't that much to deal with. Added `type: ignore` in a few places where `node.locals` would only return `node.NodeNG`. Those could theoretical be `cast` as well. Let me know what you think.